platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/TypeValidationData.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.events.problems;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.NonNullApi;
+
+import javax.annotation.Nullable;
+
+/**
+ * Additional data describing type validation problems.
+ *
+ * @since 8.9
+ */
+@Incubating
+@NonNullApi
+public interface TypeValidationData extends AdditionalData {
```

Comment:
❌ This is missing a type parameter.

Reply:
This is a Tooling API event type; it does not have the same type parameters you have it on the producer side (ie in org.gradle.api.problems.internal.TypeValidationData).

Reply:
Why do we expose the `TypeValidationData`via the TAPI? Is this something we want to have for Android Studio?

Reply:
I wanted to expose _something_ to showcase how we can expose custom additional data types. It doesn't have to be TypeValidationData, but now you mentioned it, we probably should not expose anything new on the Tooling API without a clear use-case.

Reply:
I've removed the tapi event subtype representing type validation problems.